packages/medusa/src/api/routes/store/carts/create-cart.ts

```diff
@@ -254,4 +256,7 @@ export class StorePostCartReq {
     IsOptional(),
   ])
   sales_channel_id?: string
+
+  @DbAwareColumn({ type: "jsonb", nullable: true })
```

Comment:
todo: `@DbAwareColumn` is used to define columns in database schemas. What you want to do is extend an accepted payload at the API route level, so we'll need to use the `class-validator` instead. 

The metadata is an optional object, so we should define it like so:
```
@IsOptional()
metadata?: Record<string, unknown> | undefined
```

Let me know if it makes sense.